src/async/imap/MCIMAPAsyncSession.cpp

```diff
@@ -338,19 +347,37 @@ IMAPAsyncConnection * IMAPAsyncSession::availableSession()
     }
 }
 
-IMAPAsyncConnection * IMAPAsyncSession::matchingSessionForFolder(String * folder)
+IMAPAsyncConnection * IMAPAsyncSession::sessionForFolderWithMinQueue(String * folder)
 {
-    for(unsigned int i = 0 ; i < mSessions->count() ; i ++) {
-        IMAPAsyncConnection * currentSession = (IMAPAsyncConnection *) mSessions->objectAtIndex(i);
-        if (currentSession->lastFolder() != NULL) {
-            if (currentSession->lastFolder()->isEqual(folder)) {
-                return currentSession;
+    IMAPAsyncConnection * chosenSession = NULL;
+    unsigned int minOperationsCount = std::numeric_limits<unsigned int>::max();
```

Comment:
Could you stay away from stdc++ as much as possible?
I'd rather use -1 as an indication that no value has been set yet.
(it implies casting thing but I'm fine with it)